homework/tai_nguyen/304_range_sum_query_2d.ts

```diff
@@ -0,0 +1,27 @@
+class NumMatrix {
+  prefixSum2D: number[][];
+  constructor(matrix: number[][]) {
+    this.prefixSum2D = new Array(matrix.length)
+      .fill(true)
+      .map(() => new Array(matrix[0].length).fill(0));
+
+    matrix.forEach((row, i) => {
+      row.forEach((cell, j) => {
+        this.prefixSum2D[i][j] =
+          (this.prefixSum2D[i - 1]?.[j] ?? 0) +
```

Comment:
this should work but kind of hard to read and error-prone. Perhaps consider Thinh's [implementation](https://github.com/quanghoang33/gen-6/pull/4/files#diff-0098a80934b27d3c2ee8e2a5c818073c5b8bb4e1cbaec8e7d96f1bbaa756f86eR1) or Truong's [implementation](https://github.com/quanghoang33/gen-6/pull/2/files/6175d37c6c40c2b5661a57fd74da2eb456fc89d4#diff-e290a9c1157931476d8a9f8accea220a5eb2ab375c5656c51e6c4a50b207f763R1)

Reply:
Yes, this line looks quite complex. I will fix it.
Thanks for your reference @quanghoang33